daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/preprocessing/TransactionPreprocessor.scala

```diff
@@ -62,17 +62,61 @@ private[preprocessing] final class TransactionPreprocessor(
       commands: BackStack[speedy.Command],
   ) {
     def update(
-        newCids: Iterable[ContractId],
+        newInputCids: Iterable[ContractId],
         newLocalCids: Iterable[ContractId],
         cmd: speedy.Command,
-    ): Acc = {
-      val globalCids = this.globalCids ++ newCids.filterNot(localCids)
-      if (newLocalCids.exists(globalCids))
-        fail("Conflicting discriminators between a global and local contract ID.")
-      Acc(globalCids, localCids ++ newLocalCids, commands :+ cmd)
-    }
+    ) = Acc(
+      globalCids ++ newInputCids.filterNot(localCids),
+      localCids ++ newLocalCids,
+      commands :+ cmd,
+    )
   }
 
+  /*
+   * Translates a transaction tree into a sequence of Speedy commands
+   * and collect the global contract IDs.
```

Comment:
```suggestion
   * Translates a transaction tree into a sequence of Speedy commands
   * and collects the global contract IDs.
```
Either `translates` and `collects` or both without trailing `s`.